db/db_iter.h

```diff
@@ -417,6 +418,36 @@ class DBIter final : public Iterator {
     return true;
   }
 
+  void MarkMemtableForFlushIfNeeded() {
+    if (avg_op_scan_flush_trigger_ &&
+        mem_hidden_op_since_seek_ >= memtable_op_scan_flush_trigger_ &&
+        mem_hidden_op_since_seek_ >=
+            static_cast<uint64_t>(iter_step_since_seek_) *
+                avg_op_scan_flush_trigger_) {
+      assert(memtable_op_scan_flush_trigger_ > 0);
+      active_mem_->MarkForFlush();
+      avg_op_scan_flush_trigger_ = 0;
+      memtable_op_scan_flush_trigger_ = 0;
+    }
+    iter_step_since_seek_ = 1;
+    mem_hidden_op_since_seek_ = 0;
+  }
+
+  void CheckCurKeyForFlushTrigger(uint64_t& mem_hidden_op_scanned) {
+    if (memtable_op_scan_flush_trigger_ &&
+        ikey_.sequence >= memtable_seqno_lb_) {
+      if (++mem_hidden_op_scanned >= memtable_op_scan_flush_trigger_) {
+        active_mem_->MarkForFlush();
+        // Turn off the flush trigger checks.
+        memtable_op_scan_flush_trigger_ = 0;
+        avg_op_scan_flush_trigger_ = 0;
+      }
+      if (avg_op_scan_flush_trigger_) {
+        ++mem_hidden_op_since_seek_;
```

Comment:
nit: 
(1) maybe name mem_hidden_op_scanned_since_seek_ just to align with mem_hidden_op_scanned with the main difference pointed out by "since seek"? 
(2) Indicate in CheckCurKeyForFlushTrigger() in someway that it's not just a "check". It actually does marking, reset etc. I think MarkMemtableForFlushIfNeeded() sets the right expectation